bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,220 @@
+import logging
+
+from todoist_api_python.api import TodoistAPI
+import typing_extensions
+
+from datetime import datetime, time
+
+from bugwarrior import config
+from bugwarrior.services import Service, Issue, Client
+
+log = logging.getLogger(__name__)
+
+
+class TodoistConfig(config.ServiceConfig):
+    service: typing_extensions.Literal["todoist"]
+    token: str
+    filter: str = None
+    char_open_bracket: str = "〈"
+    char_close_bracket: str = "〉"
+    inline_links: bool = True
+
+
+class TodoistClient(Client):
+    def __init__(self, token, filter):
+        self._api = TodoistAPI(token)
+        self.filter = filter
+
+    def get_projects(self):
+        all_projects = []
+        projects_iter = self._api.get_projects()
+        for projects in projects_iter:
+            for project in projects:
+                all_projects.append(project)
+        return all_projects
+
+    def get_sections(self):
+        all_sections = []
+        sections_iter = self._api.get_sections()
+        for sections in sections_iter:
+            for section in sections:
+                all_sections.append(section)
+        return all_sections
+
+    def get_users(self, project_id):
+        all_users = []
+        users_iter = self._api.get_collaborators(project_id)
+        for users in users_iter:
+            for user in users:
+                all_users.append(user)
+        return all_users
+
+    def get_issues(self):
+        tasks_iter = self._api.filter_tasks(query=self.filter)
+        return tasks_iter
```

Comment:
It would be nice to abstract away the API a bit more here.
```suggestion
        for tasks in tasks_iter:
            for task in tasks:
                yield task
```
Then you could eliminate the nested loops in the `issues` method below and just do `for issue in self.client.get_issues()`